stan/math/rev/core/var.hpp

```diff
@@ -502,6 +793,20 @@ class var_value {
     return *this;
   }
 
+  // No-op to match with Eigen methods which call eval
+  template <typename T_ = T, require_plain_type_t<T_>* = nullptr>
+  inline auto& eval() noexcept {
```

Comment:
This is missing doxygen.

Reply:
Also I believe this function should be const as well, for the same reason as above. That it also need to return a const reference (which matches the Eigen implementation)